example/ghost/t8_test_ghost_large_level_diff.cxx

```diff
@@ -0,0 +1,233 @@
+    t8_forest_set_partition(forest_partition, forest_adapt, 0);
+    t8_forest_set_ghost_ext(forest_partition, 1, T8_GHOST_FACES, 3);
+    t8_forest_set_profiling(forest_partition, 1);
+    t8_forest_commit(forest_partition);
+    if(!no_vtk){
+        t8_forest_write_vtk(forest_partition, "Forest_partition");
+    }
+    t8_global_productionf("Successfully partitioned forest.\n");
+    t8_forest_ghost_print(forest_partition);
+
+    t8_forest_print_profile (forest_partition);
+    t8_forest_unref (&forest_partition);
+
+    sc_stats_compute (comm, 1, stats);
+    sc_stats_print (t8_get_package_id (), SC_LP_STATISTICS, 1, stats, 1, 1);
+
+
+}
+
+int
+main (int argc, char *argv[])
+{
+  int                 mpiret, parsed, dim, level, refine, mpisize, helpme, no_vtk;
+  sc_options_t       *opt;
+  const char         *path;
+  char                usage[BUFSIZ];
+  char                help[BUFSIZ];
+  t8_cmesh_t          cmesh;
+
+  snprintf (usage, BUFSIZ, "Usage:\t%s <OPTIONS> <ARGUMENTS>",
+            basename (argv[0]));
+  snprintf (help, BUFSIZ,
+            "This program reads a .msh file "
+            "created by the GMSH program and constructs a "
+            "t8code coarse mesh from them.\n\n%s\n\nExample: %s -f A1\nTo open the file A1.ms."
+            "\n\nThe default dimension of the mesh to read is 2. Since the "
+            ".msh format stores elements of all (lower) dimensions "
+            "the user must provide the argument for a different dimension by hand, if "
+            "desired.\n", usage, basename (argv[0]));
+
+  mpiret = sc_MPI_Init (&argc, &argv);
+  SC_CHECK_MPI (mpiret);
+
+  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
+  p4est_init (NULL, SC_LP_ESSENTIAL);
+  t8_init (SC_LP_DEFAULT);
+
+  mpiret = sc_MPI_Comm_size (sc_MPI_COMM_WORLD, &mpisize);
+  SC_CHECK_MPI (mpiret);
+
+  opt = sc_options_new (argv[0]);
+  sc_options_add_switch(opt, 'h', "help", &helpme, "Display a short help message:");
+  sc_options_add_string (opt, 'p', "path", &path, "", "The path to the mesh-file. "
+                         "The file must end in .msh and be created with gmsh.");
+  sc_options_add_int (opt, 'd', "dim", &dim, 2, "The dimension of the mesh.");
+  sc_options_add_int (opt, 'l', "level", &level, 0, "The intial refinement level of the mesh.");
+  sc_options_add_int (opt, 'r', "refine", &refine, 0, "The number of levels that the forest "
+                       "is refined from the initial level.");
+  sc_options_add_switch(opt, 'o', "no-vtk", &no_vtk, "Suppress vtk output.");
+  parsed =sc_options_parse (t8_get_package_id (), SC_LP_ERROR, opt, argc, argv);
+  if(helpme){
+      /* display help message and usage */
+      t8_global_productionf ("%s\n", help);
+      sc_options_print_usage (t8_get_package_id (), SC_LP_ERROR, opt, NULL);
+  }
+  else if (parsed >= 0 && 0 <= level && 0 <= refine) {
+    t8_ghost_large_level_diff(path, dim, level, refine, no_vtk,sc_MPI_COMM_WORLD);
+    return 1;
```

Comment:
Remove this return.
Using it skips the sc_finalize and sc_MPI_Finalize calls.
If you test this code with > 1 process (i.e. mpirun -np 2 ./t8_test_ghost_large_level_diff) it causes an MPI error b/c
MPI is not finalized correctly.